StitchCore/StitchCore/Source/Core/StitchClient.swift

```diff
@@ -268,27 +276,41 @@ public class StitchClient: StitchClientType {
     public func login(withProvider provider: AuthProvider) -> Promise<UserId> {
         self.authProvider = provider
 
-        if isAuthenticated, let auth = auth {
-            printLog(.info, text: "Already logged in, using cached token.")
-            return Promise.init(value: auth.userId)
+        func doLoginRequest() -> Promise<UserId> {
+            return httpClient.doRequest {
+                $0.method = .post
+                $0.endpoint = self.routes.authProvidersLoginRoute(provider: provider.type)
+                $0.isAuthenticatedRequest = false
+                try $0.encode(withData: self.getAuthRequest(provider: provider))
+                }.flatMap { [weak self] any in
+                    guard let strongSelf = self else { throw StitchError.clientReleased }
+                    let authInfo = try JSONDecoder().decode(AuthInfo.self,
+                                                            from: JSONSerialization.data(withJSONObject: any))
+                    strongSelf.httpClient.authInfo = authInfo
+                    strongSelf._auth = Auth(stitchClient: strongSelf,
+                                            stitchHttpClient: strongSelf.httpClient,
+                                            userId: authInfo.userId)
+                    strongSelf.onLogin()
+                    return authInfo.userId
+                }
         }
 
-        return httpClient.doRequest {
-            $0.method = .post
-            $0.endpoint = self.routes.authProvidersLoginRoute(provider: provider.type)
-            $0.isAuthenticatedRequest = false
-            try $0.encode(withData: self.getAuthRequest(provider: provider))
-        }.flatMap { [weak self] any in
-            guard let strongSelf = self else { throw StitchError.clientReleased }
-            let authInfo = try JSONDecoder().decode(AuthInfo.self,
-                                                    from: JSONSerialization.data(withJSONObject: any))
-            strongSelf.httpClient.authInfo = authInfo
-            strongSelf._auth = Auth(stitchClient: strongSelf,
-                                    stitchHttpClient: strongSelf.httpClient,
-                                    userId: authInfo.userId)
-            strongSelf.onLogin()
-            return authInfo.userId
+        if isAuthenticated, let auth = auth {
+            if provider.type == AuthProviderTypes.anonymous.rawValue &&
+                loggedInProviderType == AuthProviderTypes.anonymous.rawValue {
```

Comment:
[note] This will have the effect of logging out an anonymous user since we aren't migrating. This results in a loss of an anonymous account which is okay during our stage of development. We should document it though.

Reply:
Would we just put that in release notes somewhere or do you want me to add a comment somewhere?

Reply:
Release notes